WordPress/wordpress-playground
# [Service Worker] Support redirects to relative URLs in Safari

Before this PR when PHP tried to make a redirect (300 response) to a relative URL, Safari would return a network error and the new page wouldn't load in Playground, we would see a white screen.

This is caused [by a Safari bug](https://bugs.webkit.org/show_bug.cgi?id=282427) which prevents Safari from redirecting responses to relative URLs if the response comes from a Service Worker.

Until [the bugfix is implemented in Safari](https://github.com/WebKit/WebKit/commit/8aeae6caf5398bb8369bc280e2acab4b8554c0af), relative URLs can only be correctly redirected using `Response.redirect`.
This [workaround was suggested in the bug report. ](https://bugs.webkit.org/show_bug.cgi?id=282427#c2)

Fixes https://github.com/WordPress/wordpress-playground/issues/645

## Testing Instructions (or ideally a Blueprint)

**Test this in Safari**

- Open[ the post editor](http://127.0.0.1:5400/website-server/?url=%2Fwp-admin%2Fpost.php%3Fpost%3D1%26action%3Dedit)
- Click on the kebab menu (upper right corner)
- Select _Preferences_ from the menu
- In the _General_ tab toggle _Custom fields_ to be active
- Click on _Show & Reload Page_
- Confirm the page reloads
